scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/Goal.java

```diff
@@ -223,6 +233,13 @@ protected Goal fill(Goal goal) {
         goal.temporalContext = new WindowsWrapperExpression(windows);
       }
 
+      if (planHorizon == null) {
```

Comment:
Maybe 
```suggestion
      Objects.requireNonNull(planHorizon);
```
is more appropriate?

Reply:
There are many examples in the code where == null was used (few right above this one). Any particular reason why you consider to use requireNonNull particularly in this case?